skrub/_table_vectorizer.py

```diff
@@ -802,7 +621,7 @@ def _auto_cast(self, X: pd.DataFrame) -> pd.DataFrame:
                     X[col] = X[col].astype(X[col].dtype.type, errors="ignore")
                 except (TypeError, ValueError):
                     pass
-            self.types_[col] = X[col].dtype
+            self.types_[col_idx] = X[col].dtype
```

Comment:
why do we want the col_idx instead of the col name?

Reply:
To not have to rely on the `columns_` fitted attribute that is not useful at the end. By removing it, we don't have the information but having the indices, we get the same results and one fitted attribute less.

Note that `columns_` is already available through the `feature_names_in_` indeed.